contracts/BondStorage.sol

```diff
@@ -18,13 +19,15 @@ contract BondStorage is AccessControl {
     uint8 public otherUsdDec;
 
     constructor(address _gatewayAddress, address _chainlinkEurOther, uint8 _otherUsdDec) {
-        _setupRole(WHITELIST_BOND_STORAGE, msg.sender);
+        _grantRole(WHITELIST_ADMIN, msg.sender);
+        _setRoleAdmin(WHITELIST_BOND_STORAGE, WHITELIST_ADMIN);
+        grantRole(WHITELIST_BOND_STORAGE, msg.sender);
```

Comment:
why mix the underscore with the safeguarded `grantRole`?

Reply:
the advice from [the docs](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/83277ff916ac4f58fec072b8f28a252c1245c2f1/docs/modules/ROOT/pages/access-control.adoc#using-accesscontrol) is to use `_grantRole` within the constructor to set up your admin permissions

then use the admin permissions to use `grantRole` where possible. it might be a bit OTT to use `grantRole` here while still in the constructor but it's possible so why not?